### DISCUSSION TITLE
Inconsistent Autocomplete for PolarsDataType

### DISCUSSION THREAD

**Discussion by stevenlis** [microsoft/pylance-release#4253] [Enhancement]
## Environment data

-   Language Server version: v2023.4.20
-   OS and version: macos 13.3 (22E252)
-   Python version (& distribution if applicable, e.g. Anaconda): 3.9.12 conda

## Code Snippet

```python
import polars as pl

pl.DataFrame(
    {'a': ['1']}
).with_columns(
    pl.col('a').cast(pl.Int16)
)
```

## Repro Steps

1. XXX

## Expected behavior

The behavior in interactive window is expected/preferred (complete without `()`):

![CS-4ZPb7Jsr](https://user-images.githubusercontent.com/33796896/232631445-cbaf6774-bea7-4364-94e3-a1152a949ec0.gif)

https://pola-rs.github.io/polars/py-polars/html/reference/expressions/api/polars.Expr.cast.html

## Actual behavior

However, in the editor, it complete with `()` 
![CS-zjOcb6PW](https://user-images.githubusercontent.com/33796896/232633054-196ec39e-ca67-4ca3-a2c6-7351d44dee9a.gif)

## Logs

[log.txt](https://github.com/microsoft/pylance-release/files/11256595/log.txt)

**Comment by rchiodo** [comment]
Thanks for the issue. 

This is on purpose at the moment. Any time you complete a class it assumes you're attempting to construct an instance of that class and automatically adds the parenthesis.

Specifically in cases where there's an assignment (like this):

```
x = pl.Int16()
```

Or it's passed as an argument to a function call (like this):

```
x = foo(pl.Int16())
```

Your example is the latter one. You can disable this by turning off the auto parens functionality:

```
"python.analysis.completeFunctionParens": false
```

**Comment by stevenlis** [comment]
I believe this will turn off `()` completion for everything, right?
```
"python.analysis.completeFunctionParens": false
```
Why does it behave differently in editor than interactive window of Jupyter extension?

**Comment by rchiodo** [comment]
> I believe this will turn off () completion for everything, right?

Yes it will.

> Why does it behave differently in editor than interactive window of Jupyter extension?

The behavior is the same for me in a python file.

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.
